### PR TITLE
compile: Move and rename `SimpleIntrinsics` -> BuiltinsContext

### PR DESCRIPTION
**Pull request by CohenArthur**
This will allow us to use the class for more than just math builtins

### CODE REVIEW

File: gcc/rust/backend/rust-compile-intrinsic.h [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #define RUST_COMPILE_INTRINSIC
 
 #include "rust-compile-context.h"
```
Comment (philberty) [inline gcc/rust/backend/rust-compile-intrinsic.h:20]: does it need the include here?

**Review by philberty — APPROVED** [review]
LGTM

File: gcc/rust/backend/rust-compile-intrinsic.h (hunk 1 above)
Comment (CohenArthur) [reply gcc/rust/backend/rust-compile-intrinsic.h:20]: You're right, I cleaned it up to include only the essentials. Good point

**Pull request merged**